ime/app/src/main/java/com/anysoftkeyboard/ime/AnySoftKeyboardBase.java

```diff
@@ -329,4 +351,62 @@ public void onCancel() {
         // the user released their finger outside of any key... okay. I have nothing to do about
         // that.
     }
+
+    @RequiresApi(Build.VERSION_CODES.R)
+    @Nullable
+    @Override
+    public InlineSuggestionsRequest onCreateInlineSuggestionsRequest(@NonNull Bundle uiExtras) {
+        Size smallestSize = new Size(0, 0);
+        Size biggestSize = new Size(Integer.MAX_VALUE, Integer.MAX_VALUE);
+
+        UiVersions.StylesBuilder stylesBuilder = UiVersions.newStylesBuilder();
+
+        InlineSuggestionUi.Style style = InlineSuggestionUi.newStyleBuilder().build();
+        stylesBuilder.addStyle(style);
+
+        Bundle stylesBundle = stylesBuilder.build();
+
+        InlinePresentationSpec spec = new InlinePresentationSpec.Builder(smallestSize, biggestSize).setStyle(stylesBundle).build();
+
+        List<InlinePresentationSpec> specList = new ArrayList<>();
+        specList.add(spec);
+
+        InlineSuggestionsRequest.Builder builder = new InlineSuggestionsRequest.Builder(specList);
+
+        return builder.setMaxSuggestionCount(InlineSuggestionsRequest.SUGGESTION_COUNT_UNLIMITED).build();
+    }
+
+    @RequiresApi(Build.VERSION_CODES.R)
+    @Override
+    public boolean onInlineSuggestionsResponse(@NonNull InlineSuggestionsResponse response) {
+        List<InlineSuggestion> inlineSuggestions = response.getInlineSuggestions();
+
+        if (mInputViewContainer == null) {
+            getMainExecutor().execute(() -> mInputViewContainer = createInputViewContainer());
+        }
+
+        if (mInputViewContainer != null) {
+            LinearLayout inlineAutofillLayout = mInputViewContainer.getInlineAutofillView();
+            float height = TypedValue.applyDimension(
+                    TypedValue.COMPLEX_UNIT_DIP,
+                    40,
+                    getResources().getDisplayMetrics()
+            );
+            Size autofillSize = new Size(ViewGroup.LayoutParams.WRAP_CONTENT, ((int) height));
+
+            Executor executor = Executors.newSingleThreadExecutor();
+
+            getMainExecutor().execute(inlineAutofillLayout::removeAllViews);
+            
+            for (InlineSuggestion inlineSuggestion : inlineSuggestions) {
+                try {
+                    inlineSuggestion.inflate(this, autofillSize, executor, inlineContentView -> getMainExecutor().execute(() -> inlineAutofillLayout.addView(inlineContentView)));
```

Comment:
Why do you need to do this in a `execute` block? I assume this method is called in the main thread, no?

Reply:
I am unsure where this is actually called from given that the documentation was poor and I was having a very hard time getting the views to show given the exceptionally poor documentation. I can try removing the main executor and see if that has any effect though

Reply:
swapping to main executor was a success, thanks for the great note!

Reply:
YAY! Less thread switching is easier to read